# STAT 207 Lab Assignment 10 - Group Part - [9 Points]



## <u>Case Study</u>: Predicting Airbnb Superhost Status

Suppose that you have a spare room in your house (in Seattle) and you are considering advertising this room to guests on Airbnb. Airbnb superhosts are considered to be experienced hosts who provide a shining example for other hosts, and extraordinary experiences for their guests. Once a host reaches Superhost status, a superhost badge will automatically appear on their listing and profile to help guests identify them. You would like to assess your chances of being named a superhost with your property.

The following dataset is a sample (assume random) of available Airbnb listings in Seattle, WA. These listings were collected in January 2016, and filtered to just contain listings from the five most popular Seattlbe neighborhoods (for Airbnb listings) and just contain listings that are either in a house or apartment property.

This dataset contains the following variables.

**Listing Information**
The dataset contains the following information about the Airbnb *listing*:
* <u>price</u>: price of the listing (per night)(in US dollars)
* <u>review_scores_rating</u>: the average rating of the listing [0,100] (100 is the best)
* <u>number_of_reviews</u>: the number of reviews for the listing
* <u>security_deposit</u>: the security deposit required for the listing (in US dollars)
* <u>cleaning_fee</u>: the cleaning fee required for the listing (in US dollars)
* <u>neighborhood</u>: the neighborhood of Seattle the listing is located in
* <u>property_type</u>: is the listing in a 'House' or 'Apartment'
* <u>room_type</u>: is the listing a 'Entire home/apt', 'Private room', or 'Shared room'
* <u>accommodates</u>: how many guests will the listing accommodate
* <u>bathrooms</u>:how many bathrooms does the listing have
* <u>beds</u>: how many beds does the listing have

**Host Information**
The dataset also contains the following information about the *host* of the given Airbnb listing:
* <u>host_is_superhost</u>: is the host a "superhost": t=True, f=False
* <u>host_has_profile_pic</u>: does the host have a profile pic in their bio: t=True, f=False
* <u>host_response_time</u>: how fast will the host respond to requests (on average)

## Research Goals

The primary research goal that we will pursue in this lab will be to build a predictive model that effectively predicts the `probability a Seattle Airbnb host is a **superhost**` for *new Seattle Airbnb listings/hosts* given some combination of the `remaining explanatory variables`.



## Points

<p>&nbsp;</p>
<table style="border: none;border-collapse: collapse;width:102pt;">
    <tbody>
        <tr>
            <td style="color:black;font-size:15px;font-weight:700;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:general;vertical-align:bottom;border:.5pt solid black;height:14.25pt;width:51pt;">Problem</td>
            <td style="color:black;font-size:15px;font-weight:700;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:general;vertical-align:bottom;border:.5pt solid black;border-left:none;width:51pt;">Points</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">1.1</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">0.25</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">1.2</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">0.75</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">1.3</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">0.5</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">2.1</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">1</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">2.2</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">0.75</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">2.3</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">0.25</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">2.4</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">1</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">3</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">3</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">4</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">1.5</td>
        </tr>
    </tbody>
</table>

In [6]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.linear_model import Ridge
from sklearn.linear_model import ElasticNet
from sklearn.preprocessing import StandardScaler
import statsmodels.formula.api as smf
from sklearn.linear_model import LogisticRegression

## 1. Basic Data Examination and Manipulation

### 1.1. Basics
1. Read this csv file into a dataframe.

2. Display your dataframe.

3. Show how many rows your dataframe has.



In [7]:
seattle_df = pd.read_csv('seattle_airbnb_listings_cleaned.csv')
seattle_df

price  review_scores_rating  number_of_reviews  security_deposit  \
0      300                   100                 24               500   
1      149                    96                 11               300   
2       95                    95                 79               150   
3      105                   100                 13               500   
4      140                    99                 30               250   
..     ...                   ...                ...               ...   
334    250                    95                  4               250   
335    125                    95                 12               150   
336    100                    98                 10               200   
337    175                    96                 29               500   
338    154                    92                  5               150   

     cleaning_fee neighborhood property_type        room_type  accommodates  \
0              95  Wallingford         House  Entire home/apt             5   
1             105  Wallingford     Apartment  Entire home/apt             6   
2              40  Wallingford     Apartment  Entire home/apt             3   
3              50  Wallingford         House     Private room             2   
4              65  Wallingford         House  Entire home/apt             2   
..            ...          ...           ...              ...           ...   
334           100      Fremont         House  Entire home/apt            16   
335            45      Fremont     Apartment  Entire home/apt             2   
336            89      Fremont     Apartment  Entire home/apt             6   
337            95      Fremont         House  Entire home/apt             6   
338            85      Fremont         House  Entire home/apt             5   

     bathrooms  beds  host_response_time  host_response_rate  \
0          1.5     3  within a few hours                1.00   
1          1.0     3      within an hour                1.00   
2          1.0     2      within an hour                1.00   
3          2.0     1  within a few hours                1.00   
4          1.0     1      within an hour                1.00   
..         ...   ...                 ...                 ...   
334        3.5    10  within a few hours                0.97   
335        1.0     1  within a few hours                1.00   
336        1.0     2      within an hour                1.00   
337        2.0     3  within a few hours                1.00   
338        1.0     3      within an hour                1.00   

    host_is_superhost host_has_profile_pic host_identity_verified  
0                   t                    t                      t  
1                   f                    t                      t  
2                   f                    t                      t  
3                   t                    t                      t  
4                   t                    t                      t  
..                ...                  ...                    ...  
334                 f                    t                      t  
335                 t                    t                      t  
336                 f                    t                      t  
337                 f                    t                      t  
338                 f                    t                      t  

[339 rows x 16 columns]

In [8]:
print('Rows:',seattle_df.shape[0])

Rows: 339


### 1.2. 0/1 Response Variable

Create a new 0/1 variable in your dataset in which `superhosts=1` and `nonsuperhosts=0`. 

In [9]:
seattle_df['is_superhost'] = (seattle_df['host_is_superhost'] == 't').astype(int)
seattle_df

price  review_scores_rating  number_of_reviews  security_deposit  \
0      300                   100                 24               500   
1      149                    96                 11               300   
2       95                    95                 79               150   
3      105                   100                 13               500   
4      140                    99                 30               250   
..     ...                   ...                ...               ...   
334    250                    95                  4               250   
335    125                    95                 12               150   
336    100                    98                 10               200   
337    175                    96                 29               500   
338    154                    92                  5               150   

     cleaning_fee neighborhood property_type        room_type  accommodates  \
0              95  Wallingford         House  Entire home/apt             5   
1             105  Wallingford     Apartment  Entire home/apt             6   
2              40  Wallingford     Apartment  Entire home/apt             3   
3              50  Wallingford         House     Private room             2   
4              65  Wallingford         House  Entire home/apt             2   
..            ...          ...           ...              ...           ...   
334           100      Fremont         House  Entire home/apt            16   
335            45      Fremont     Apartment  Entire home/apt             2   
336            89      Fremont     Apartment  Entire home/apt             6   
337            95      Fremont         House  Entire home/apt             6   
338            85      Fremont         House  Entire home/apt             5   

     bathrooms  beds  host_response_time  host_response_rate  \
0          1.5     3  within a few hours                1.00   
1          1.0     3      within an hour                1.00   
2          1.0     2      within an hour                1.00   
3          2.0     1  within a few hours                1.00   
4          1.0     1      within an hour                1.00   
..         ...   ...                 ...                 ...   
334        3.5    10  within a few hours                0.97   
335        1.0     1  within a few hours                1.00   
336        1.0     2      within an hour                1.00   
337        2.0     3  within a few hours                1.00   
338        1.0     3      within an hour                1.00   

    host_is_superhost host_has_profile_pic host_identity_verified  \
0                   t                    t                      t   
1                   f                    t                      t   
2                   f                    t                      t   
3                   t                    t                      t   
4                   t                    t                      t   
..                ...                  ...                    ...   
334                 f                    t                      t   
335                 t                    t                      t   
336                 f                    t                      t   
337                 f                    t                      t   
338                 f                    t                      t   

     is_superhost  
0               1  
1               0  
2               0  
3               1  
4               1  
..            ...  
334             0  
335             1  
336             0  
337             0  
338             0  

[339 rows x 17 columns]

### 1.3. Train-Test-Split

Create a training dataset and a test dataset from your dataframe using a random state of `101` and a test dataset size that is a random sample of about 20% of the observations from the original dataset.

In [10]:
seattle_test,seattle_train = train_test_split(seattle_df,test_size=0.2,random_state=101)

## 2. Full Model and Test Log-Likelihood Function

In the interest of time, let's only consider using 5 potential explanatory variables in our logistic regression model, which will predict the `probability that a host is a superhost`:

* `review_scores_rating`
* `number_of_reviews`
* `neighborhood`
* `room_type`
* `property_type`

### 2.1. Fit the Model

Fit this "full" logistic regression model and show the summary output table. Use this summary output table to find the log-likelihood function value of this model and the training dataset.

In [11]:
full_model = smf.ols(data=seattle_train,formula='is_superhost~review_scores_rating+number_of_reviews+neighborhood+room_type+property_type').fit()
full_model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:           is_superhost   R-squared:                       0.279
Model:                            OLS   Adj. R-squared:                  0.167
Method:                 Least Squares   F-statistic:                     2.489
Date:                Tue, 31 Oct 2023   Prob (F-statistic):             0.0177
Time:                        23:30:13   Log-Likelihood:                -35.172
No. Observations:                  68   AIC:                             90.34
Df Residuals:                      58   BIC:                             112.5
Df Model:                           9                                         
Covariance Type:            nonrobust                                         
===============================================================================================
                                  coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------------------
Intercept                      -0.6137      0.472     -1.300      0.199      -1.559       0.331
neighborhood[T.Broadway]       -0.3906      0.159     -2.453      0.017      -0.709      -0.072
neighborhood[T.Fremont]        -0.5254      0.215     -2.442      0.018      -0.956      -0.095
neighborhood[T.Minor]          -0.3281      0.221     -1.484      0.143      -0.771       0.114
neighborhood[T.Wallingford]    -0.1485      0.193     -0.770      0.444      -0.535       0.238
room_type[T.Private room]      -0.1398      0.168     -0.832      0.409      -0.476       0.197
room_type[T.Shared room]       -0.5289      0.276     -1.917      0.060      -1.081       0.024
property_type[T.House]          0.0649      0.149      0.437      0.664      -0.232       0.362
review_scores_rating            0.0126      0.005      2.532      0.014       0.003       0.023
number_of_reviews               0.0023      0.001      1.580      0.120      -0.001       0.005
==============================================================================
Omnibus:                        9.145   Durbin-Watson:                   1.626
Prob(Omnibus):                  0.010   Jarque-Bera (JB):                3.159
Skew:                           0.139   Prob(JB):                        0.206
Kurtosis:                       1.981   Cond. No.                         898.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Log-Likelihood:	-35.172

### 2.2. Training Data Predictive Probabilities

Calculate the predictive probabilities for all observations in the **training** dataset with model. **Call the pandas series that contains each of these predictive probabilities `phat_train`**.

*Note: In addition to using the **.predict()** function to predict the **predictive probability** of a single observation with a logistic regression model, we can use **.predict(YOUR_DATAFRAME)** to predict the predictive probabilities for all observations in a given dataframe with a logistic regression model.*

In [12]:
phat_train = full_model.predict(seattle_train)
phat_train

172    0.063449
38     0.570393
242    0.151654
64     0.591473
181    0.424916
         ...   
157    0.780811
247    0.405172
70     0.277378
104    0.391359
312    0.084234
Length: 68, dtype: float64

### 2.3. Training Log-Likelihood Function Value

Assuming that your training dataframe was called `df_train`, the code below should calculate the the log-likelihood function value for your training dataset and this model. (If not, modify the code below to contain the actual name of your training dataframe).

This **log_loss()** should take take the following values:
* the actual 0/1 response variable values
* the predictive probabilities of a given model

Verify that this output here is the same as what you got in 2.1.

In [22]:
from sklearn.metrics import log_loss

-log_loss(seattle_train['host_is_superhost'], phat_train)*len(seattle_train['host_is_superhost'])

-35.17220687267453


### 2.4. Test Log-Likelihood Function Value

Use the code in 2.2 and 2.3 to now calculate the log-likelihood function value of the **test dataset** and this model.

In [14]:
phat_test = full_model.predict(seattle_test)

log_loss(seattle_test['host_is_superhost'], phat_test)*len(seattle_test['host_is_superhost'])

154.46748342826666

## 3. Backwards Elimination Algorithm

Ideally, we'd like to train (with our training dataset) and select the logistic regression model that will yield the HIGHEST **test log-likelihood function (LLF) value** (in our case it will be the LLF that is the least negative). Let's use a backwards elimination algorithm to try to help us find this model with the highest test LLF value.

*Note: Remember that the backwards elimination algorithm is just an algorithm "template". You can use this algorithm template to evaluate other types of models (like logistic regression) in attempt to find the model that has the highest metric value (like the highest test LLF value).*

In [29]:
current = smf.logit(formula='is_superhost~review_scores_rating+number_of_reviews+neighborhood+room_type+property_type', data=seattle_train).fit()
current_test = current.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],current_test)*len(seattle_test['is_superhost'])

Optimization terminated successfully.
         Current function value: 0.429991
         Iterations 31


-182.38991426323844

In [16]:
#remove review-scores-rating
remove_scores_rating = ('is_superhost~number_of_reviews+neighborhood+room_type+property_type')
remove_scores_mod = smf.logit(formula=remove_scores_rating, data=seattle_train).fit()
score_test = remove_scores_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],score_test)*len(seattle_test['is_superhost'])

         Current function value: 0.537145
         Iterations: 35


/Users/sabarishmogallapalli/miniconda3/lib/python3.11/site-packages/statsmodels/base/model.py:607: ConvergenceWarning: Maximum Likelihood optimization failed to converge. Check mle_retvals
  warnings.warn("Maximum Likelihood optimization failed to "


-183.89997164227808

In [19]:
#remove number_of_reviews

remove_number_reviews = ('is_superhost~review_scores_rating+neighborhood+room_type+property_type')
remove_number_reviews_mod = smf.logit(formula=remove_number_reviews,data=seattle_train).fit()
reviews_test = remove_number_reviews_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],reviews_test)*len(seattle_test['is_superhost'])

         Current function value: 0.473117
         Iterations: 35


/Users/sabarishmogallapalli/miniconda3/lib/python3.11/site-packages/statsmodels/base/model.py:607: ConvergenceWarning: Maximum Likelihood optimization failed to converge. Check mle_retvals
  warnings.warn("Maximum Likelihood optimization failed to "


-181.8382384022453

In [24]:
#remove neighborhood

remove_neighborhood = ('is_superhost~review_scores_rating+number_of_reviews+room_type+property_type')
remove_neighborhood_mod = smf.logit(formula=remove_neighborhood,data=seattle_train).fit()
neighborhood_test = remove_neighborhood_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],neighborhood_test)*len(seattle_test['is_superhost'])

         Current function value: 0.464800
         Iterations: 35


/Users/sabarishmogallapalli/miniconda3/lib/python3.11/site-packages/statsmodels/base/model.py:607: ConvergenceWarning: Maximum Likelihood optimization failed to converge. Check mle_retvals
  warnings.warn("Maximum Likelihood optimization failed to "


-170.78414262930775

In [26]:
#remove roomtype
remove_roomtype = ('is_superhost~review_scores_rating+number_of_reviews+neighborhood+property_type')
remove_roomtype_mod = smf.logit(formula=remove_roomtype,data=seattle_train).fit()
roomtype_test = remove_roomtype_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],roomtype_test)*len(seattle_test['is_superhost'])

Optimization terminated successfully.
         Current function value: 0.465017
         Iterations 8


-143.5904561192218

In [27]:
#remove propertytype
remove_propertytype = ('is_superhost~review_scores_rating+number_of_reviews+neighborhood+room_type')
remove_propertytype_mod = smf.logit(formula=remove_propertytype,data=seattle_train).fit()
propertytype_test = remove_propertytype_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],propertytype_test)*len(seattle_test['is_superhost'])

         Current function value: 0.431900
         Iterations: 35


/Users/sabarishmogallapalli/miniconda3/lib/python3.11/site-packages/statsmodels/base/model.py:607: ConvergenceWarning: Maximum Likelihood optimization failed to converge. Check mle_retvals
  warnings.warn("Maximum Likelihood optimization failed to "


-172.45816167881264

In [30]:
#remove roomtype - CURRENT 2
current_two = ('is_superhost~review_scores_rating+number_of_reviews+neighborhood+property_type')
current_two_mod = smf.logit(formula=current_two,data=seattle_train).fit()
current_two_test = current_two_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],current_two_test)*len(seattle_test['is_superhost'])

Optimization terminated successfully.
         Current function value: 0.465017
         Iterations 8


-143.5904561192218

In [31]:
#remove scores
remove_scores = ('is_superhost~number_of_reviews+neighborhood+property_type')
remove_scores_mod = smf.logit(formula=remove_scores,data=seattle_train).fit()
remove_scores_test = remove_scores_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],remove_scores_test)*len(seattle_test['is_superhost'])

Optimization terminated successfully.
         Current function value: 0.565952
         Iterations 6


-164.40323041099353

In [33]:
#remove reviews
remove_reviews = ('is_superhost~review_scores_rating+neighborhood+property_type')
remove_reviews_mod = smf.logit(formula=remove_reviews,data=seattle_train).fit()
remove_reviews_test = remove_reviews_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],remove_reviews_test)*len(seattle_test['is_superhost'])

Optimization terminated successfully.
         Current function value: 0.512114
         Iterations 8


-155.8096905201557

In [32]:
#remove neighborhood
remove_neighborhood = ('is_superhost~number_of_reviews+review_scores_rating+property_type')
remove_neighborhood_mod = smf.logit(formula=remove_neighborhood,data=seattle_train).fit()
remove_neighborhood_test = remove_neighborhood_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],remove_neighborhood_test)*len(seattle_test['is_superhost'])

Optimization terminated successfully.
         Current function value: 0.502913
         Iterations 8


-142.43272473949014

In [34]:
#remove propertytype
remove_propertytype = ('is_superhost~number_of_reviews+review_scores_rating+neighborhood')
remove_propertytype_mod = smf.logit(formula=remove_propertytype,data=seattle_train).fit()
remove_propertytype_test = remove_propertytype_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],remove_propertytype_test)*len(seattle_test['is_superhost'])

Optimization terminated successfully.
         Current function value: 0.465225
         Iterations 8


-143.9244874668851

In [35]:
#remove neighborhood - CURRENT 3
current_three = ('is_superhost~number_of_reviews+review_scores_rating+property_type')
current_three_mod = smf.logit(formula=current_three,data=seattle_train).fit()
current_three_test = current_three_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],current_three_test)*len(seattle_test['is_superhost'])

Optimization terminated successfully.
         Current function value: 0.502913
         Iterations 8


-142.43272473949014

In [36]:
#remove reviews
remove_reviews = ('is_superhost~review_scores_rating+property_type')
remove_reviews_mod = smf.logit(formula=remove_reviews,data=seattle_train).fit()
remove_reviews_test = remove_reviews_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],remove_reviews_test)*len(seattle_test['is_superhost'])

Optimization terminated successfully.
         Current function value: 0.557398
         Iterations 8


-147.76606918759705

In [37]:
#remove scores
remove_scores = ('is_superhost~number_of_reviews+property_type')
remove_scores_mod = smf.logit(formula=remove_scores,data=seattle_train).fit()
remove_scores_test = remove_scores_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],remove_scores_test)*len(seattle_test['is_superhost'])

Optimization terminated successfully.
         Current function value: 0.622842
         Iterations 5


-161.71033792562434

In [38]:
#remove propertytype
remove_propertytype = ('is_superhost~number_of_reviews+review_scores_rating')
remove_propertytype_mod = smf.logit(formula=remove_propertytype,data=seattle_train).fit()
remove_propertytype_test = remove_propertytype_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],remove_propertytype_test)*len(seattle_test['is_superhost'])

Optimization terminated successfully.
         Current function value: 0.508411
         Iterations 8


-141.5269437257977

In [39]:
#remove propertytype - CURRENT 4
current_four = ('is_superhost~number_of_reviews+review_scores_rating')
current_four_mod = smf.logit(formula=current_four,data=seattle_train).fit()
current_four_test = current_four_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],current_four_test)*len(seattle_test['is_superhost'])

Optimization terminated successfully.
         Current function value: 0.508411
         Iterations 8


-141.5269437257977

In [40]:
#remove reviews
remove_reviews = ('is_superhost~review_scores_rating')
remove_reviews_mod = smf.logit(formula=remove_reviews,data=seattle_train).fit()
remove_reviews_test = remove_reviews_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],remove_reviews_test)*len(seattle_test['is_superhost'])

Optimization terminated successfully.
         Current function value: 0.558426
         Iterations 8


-148.14091038117996

In [41]:
#remove rating
remove_rating = ('is_superhost~number_of_reviews')
remove_rating_mod = smf.logit(formula=remove_rating,data=seattle_train).fit()
remove_rating_test = remove_rating_mod.predict(seattle_test)

-log_loss(seattle_test['is_superhost'],remove_rating_test)*len(seattle_test['is_superhost'])

Optimization terminated successfully.
         Current function value: 0.629816
         Iterations 5


-160.97870044533744

## 4. Final Model

Write out your final logistic regression model in equation form using the three different representations that we discussed in class.
1. The one that predicts the **probability** the host is a superhost on the left hand side.
2. The one that predicts the **odds** the host is a superhost on the left hand side.
3. The one that predicts the **log-odds** the host is a superhost on the left hand side.

Make sure that you use the appropriate notation that we discussed in class.